### DISCUSSION TITLE
How do I turn off the auto close feature for tags in Phoenix Code?

### DISCUSSION THREAD

**Discussion by annasassen** [phcode-dev/phoenix#1990] [Q&A]
Hello, I am new to HTML and I really like Phoenix Code, but the auto close feature for tags is really irritating me. I've been advised that as a newbie it's best to close tags manually so you build the muscle memory. I have searched the entire internet looking for how to turn this off, I've had ChatGPT help me, but I cannot seem to find where to turn it off!

**Comment by devvaannsh** [comment]
Hey, yes you can turn this feature off. Just open the preferences file and paste this 
```
    "closeTags": {
        "whenClosing": false,
        "whenOpening": false
    }
```

You can refer to [this docs](https://docs.phcode.dev/docs/editing-text#editing-preferences) if you need any help on how to modify the preferences.

**Reply by annasassen** [reply]
Thank you! I think that worked.... buuuut there's a new problem. It no longer auto populates the closing tag when I write the start tag, but when I start to write the closing tag as soon as I hit the forward slash, it auto populates the rest of the closing tag. I am sorry for being such a n00b. :(

**Reply by devvaannsh** [reply]
That should not be the case. Looks like a bug to me, because `whenClosing: false` was set to **NOT** auto populate the rest of the closing tag. I tried it on my machine and it doesn't auto-populate the whole tag. 
DW, our team will look into it. But till then one workaround is to just write the commonly used tags in `dontCloseTags` option.
```
    "closeTags": {
        "whenClosing": false,
        "whenOpening": false,
        "dontCloseTags": ["body", "html", "a", "span", "div"]
    }
```

**Comment by annasassen** [comment]
I'll give that a try. Thank you! I'm pretty sure a lot of this is user
error. Lol. I'm new to all of this so I'm not familiar with all the jargon
for the settings or how to edit preferences via the editor's code. But, it
seems that HTML and CSS are easier to learn than the editor itself!
Thank you again for the help! ^_^

**Reply by devvaannsh** [reply]
Most welcome! For any editor related stuff, you can refer to the documentation site. It pretty much covers everything. [https://docs.phcode.dev/](https://docs.phcode.dev/)

**Comment by annasassen** [comment]
I did see that page, but it doesn't make much sense to me yet. It's not the
most beginner friendly. 😅

**Reply by devvaannsh** [reply]
No worries! Our team is always here to help. Also don't forget to spread the word about Phoenix Code! 😉

**Comment by annasassen** [comment]
It's me again. I am still having issues. I pasted in all the code you gave me, saved it, and restarted, but upon restart I get an error message saying the code is broken and then it reverts to the original settings. The solutions you've given me to fix the auto complete stuff requires me to edit code, and I don't know how to do that, I mean, I'm just learning the basics of HTML right now. Lol.

**Reply by devvaannsh** [reply]
Hey, would it be possible for you to share a screenshot?
